### PR TITLE
Adds conversion between R1 .mat files and Python `Project`s

### LINKED ISSUE #46: Read in RASCAL1 and RAT input files
The routines `sp.io.loadmat` and `sp.io.savemat` (https://docs.scipy.org/doc/scipy/reference/generated/scipy.io.loadmat.html#scipy.io.loadmat) allow MATLAB files to be read into python. We should make use of these routines to adapt the `r1toProjectClass` for the python code, and add a routine converting a RAT projectClass to a python project.

### PR DESCRIPTION
**Pull request by alexhroom**
Fixes #46 by adding functions `r1_to_project_class` and `project_class_to_r1`, which convert between R1 .mat files and Python `Project`s.

### CODE REVIEW

**Review by DrPaulSharp — COMMENTED** [review]
This is looking good, alongside the specific comments:

- Please make a python script to go alongside the jupyter notebook for the example.
- Have a look at the matlab example and see what you can incorporate from it into the python script and notebook.
- See if you can test the save to matlab - take a look at tests in "test_inputs" and "test_wrappers" to see if you can write somethimg similar.

File: RATapi/utils/convert.py [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+                mat_project["dataLimits"],
+                mat_project["simLimits"],
+            )
+        ]
+    )
+
+    # contrast names may be java strings (unsure why, maybe GUI input?): convert to Python str
+    # indexing gets the byte data out of the MatlabOpaque object
+    if len(mat_project["contrastNames"]) == 1 and isinstance(mat_project["contrastNames"], MatlabOpaque):
+        mat_project["contrastNames"] = bytes(mat_project["contrastNames"][0][3][7:]).decode("ascii")
```
[the remark this answers is not in the document]
Comment (alexhroom) [reply RATapi/utils/convert.py:146]: will make the comment more elaborate: the names are a `MatlabOpaque` object (wrapper class for unknown matlab types), which are given as a 1-item array with 4 entries - the first, second, third are some class info, the fourth is the actual data. then the actual data is the byte data of a Java string, which has 6 metadata bytes and then the actual string in ascii format - writing it out now i think it's definitely important to add that as a comment!

File: RATapi/utils/convert.py [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+        backgrounds=backs,
+        background_parameters=back_params,
+        bulk_in=bulk_ins,
+        bulk_out=bulk_outs,
+        scalefactors=scale_facs,
+        data=data,
+        layers=layers,
+        contrasts=contrasts,
+        custom_files=custom_file,
+    )
```
[the remark this answers is not in the document]
Comment (alexhroom) [reply RATapi/utils/convert.py:235]: the fields of Project which are not accounted for are:
`calculation`, `absorption`, `domain_ratios`, `domain_contrasts`.
I guess the only non-domain related one here would be `absorption`.

[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply RATapi/utils/convert.py:235]: Ok, that should be fine as `absorption` is a new feature in RAT and the default is `False`.

File: RATapi/utils/convert.py [hunk 3]
```diff
@@ -0,0 +1,498 @@
...
+
+    if return_struct:
+        return r1
+
+    if Path(filename).suffix != ".mat":
+        filename += ".mat"
+    # scipy.io.savemat doesn't do cells properly:
+    # https://github.com/scipy/scipy/issues/3756
+    # rather than fiddling we just use matlab
+    eng = wrappers.start_matlab().result()
```
Comment (DrPaulSharp) [inline RATapi/utils/convert.py:494]: Please add a `try-except` here to give a better error message than "'NoneType' object has no attribute 'result'"

**Review by DrPaulSharp — COMMENTED** [review]
See the specific comment, plus also please make the new example importable by adding an `__init__.py` to the convert_rascal_project directory and adding this example to the `__init__.py` in the examples directory.

**Pull request merged**